client/transaction-pool/graph/src/validated_pool.rs

```diff
@@ -505,14 +504,14 @@ impl<B: ChainApi> ValidatedPool<B> {
 	///
 	/// Consumers of this stream should use the `ready` method to actually get the
 	/// pending transactions in the right order.
-	pub fn import_notification_stream(&self) -> EventStream<ExHash<B>> {
+	pub fn import_notification_stream(&self) -> EventStream<BlockHash<B>> {
```

Comment:
this is weird change, because it's a extrinsic hash, not block hash (stream of events each containing transaction hash)

Reply:
I just used what was there. It always used the block hash. Basically this type alias just means to extract the `Block::Hash` type from the `ChainApi` trait.

Reply:
We always hashed transactions with the same hashing algorithm that is used by the block.

Reply:
I know that technically nothing changed, but it looks weird now. You can use just "Hash" for this or something.

> > Cannot we prepare all of those retracted blocks in advance and just add them to the current event to the retracted list?
> 
> Where should that be done?

The place where `ChainEvent::NewBlock` is instantiated:
https://github.com/paritytech/substrate/pull/6123/files#diff-240c17a1e4dbce4ad137b7777bf03ce4R1050

> 
> > We request them from transaction pool every time anyway, don't we?
> 
> What do we request from the transaction pool?

List of blocks whose transaction are going to be resubmitted

Reply:
I know, it just look weird and does not help reading the code :)

Something that denotes transaction hash is named `BlockHash`

Reply:
For me `ExHash` did also not help to read the code. I did not really thought about what it means. We can just call it `HashOf` or whatever.

Reply:
There reason for separate typedef was to introduce a semantic meaning and to be able to change it in the future without affecting the whole code.

IMHO we should revert this change - if `ExHash` is unclear (which I agree it is), we should rename it to `ExtrinsicHash`, but frankly this has nothing to do with the description of this PR.

Reply:
> There reason for separate typedef was to introduce a semantic meaning and to be able to change it in the future without affecting the whole code.

I can revert this change, but it was already everywhere set to `Block::Hash`. So, even if you would change it, you would need to touch all these files. Is there any reason to hash a transaction differently?

Reply:
Even if there are no reasons to use different hashing scheme for transactions, the type def gives it much better semantic meaning, i.e.:
```
fn transaction_hash(&self) -> Result<TransactionHash<B>>
// vs
fn transaction_hash(&self) -> Result<BlockHash<B>>
```
the above just reads much better, even if the underlying type is exactly the same - you don't get confused by "Why does this function return a block hash when it says 'transaction hash'"?

I'm ok with other changes in this PR and feel that every comment in that discussion is kind of wasted, cause it doesn't even remotely touch the main purpose of this PR, so I strongly propose to split this change into it's own PR where we can discuss semantics and philosophy and get the important stuff merged :)

Reply:
```
fn transaction_hash(&self) -> Result<TransactionHash<B>>
// vs
fn transaction_hash(&self) -> Result<BlockHash<B>>
```

That is fine and I will revert to `TransactionHash<B>`. I should have expressed more directly.